File: graphedu/common/models/vo/educationv2/chapter_resource.py
"""章节资源 VO 模型。"""
from datetime import datetime

from pydantic import Field

from graphedu.common.models.vo import VO
from graphedu.common.models.vo.systemv2.upload import FileInfoVO


class ChapterResourceBatchDeleteResultVO(VO):
    """章节资源批量删除结果 VO"""

    success_count: int = Field(description="成功删除数量")
    fail_count: int = Field(description="失败数量")
    results: list[dict] = Field(description="详细结果列表")


class ChapterResourceListVO(VO):
    """章节资料列表项 VO"""

    resource_id: int = Field(description="资料ID")
    chapter_id: int = Field(description="所属章节ID")
    resource_name: str = Field(description="资料名称")
    resource_type: str = Field(description="资料类型（video视频/document文档/text文本）")
    file_id: int | None = Field(default=None, description="文件ID（引用sys_upload.file_id）")
    resource_url: str | None = Field(default=None, description="外部链接URL")
    description: str | None = Field(default=None, description="描述")
    resource_data: dict | None = Field(default=None, description="扩展数据（JSONB格式）")
    # 解析相关字段（用于列表展示）
    parse_status: str | None = Field(default=None, description="解析状态（0待处理 1处理中 2处理成功 3处理失败）")
    display_order: int = Field(default=0, description="显示顺序")
    is_visible: str = Field(description="是否可见（Y/N）")
    status: str = Field(description="状态（0正常 1停用 2已删除）")
    create_time: datetime | None = Field(default=None, description="创建时间")

    # 关联信息
    file_url: str | None = Field(default=None, description="文件URL")
    file_info: FileInfoVO | None = Field(default=None, description="文件上传信息（sys_upload）")


class ChapterResourceDetailVO(ChapterResourceListVO):
    """章节资料详细信息 VO"""

    text_file_id: int | None = Field(default=None, description="纯文本文件ID")
    create_by: int | None = Field(default=None, description="创建者")
    update_by: int | None = Field(default=None, description="更新者")
    update_time: datetime | None = Field(default=None, description="更新时间")


class ChapterResourceParseSubmitVO(VO):
    """提交章节资料 PDF 解析任务响应 VO"""

    resource_id: int = Field(description="资料ID")
    mineru_task_id: str = Field(description="MinerU 任务ID")
    parse_status: str = Field(description="解析状态")


class ChapterResourceParseStatusVO(VO):
    """章节资料解析与 GraphRAG 状态 VO"""

    resource_id: int = Field(description="资料ID")
    parse_status: str = Field(description="PDF 解析状态：0待处理 / 1处理中 / 2处理成功 / 3处理失败")
    mineru_task_id: str | None = Field(default=None, description="MinerU 任务ID")
    text_file_id: int | None = Field(default=None, description="纯文本文件ID")
    page_count: int | None = Field(default=None, description="解析页数")
    markdown_length: int | None = Field(default=None, description="Markdown 内容长度")
    markdown_s3_key: str | None = Field(default=None, description="Markdown 结果对象存储 Key")
    markdown_url: str | None = Field(default=None, description="Markdown 临时访问链接")
    error_message: str | None = Field(default=None, description="错误信息")




class CeleryTaskVO(VO):
    """Celery 异步任务提交响应 VO"""

    celery_task_id: str = Field(description="Celery 任务ID")
    status: str = Field(description="提交状态，固定为 'submitted'")
    document_id: int | None = Field(default=None, description="关联文档ID（build-graph 时）")
    chapter_id: int | None = Field(default=None, description="关联章节ID（generate-description 时）")

File: graphedu-ui/src/types/api/education/chapterResource.ts
export interface ChapterResourceParseSubmitVO {
  resourceId: number
  mineruTaskId: string
  parseStatus: string
}

export interface ChapterResourceParseStatusVO {
  resourceId: number
  parseStatus: string
  mineruTaskId?: string
  textFileId?: number
  pageCount?: number
  markdownLength?: number
  markdownS3Key?: string
  markdownUrl?: string
  errorMessage?: string
}

/**
 * 章节资源批量删除结果 VO
 */
export interface ChapterResourceBatchDeleteResultVO {
  /** 成功删除数量 */
  successCount: number
  /** 失败数量 */
  failCount: number
  /** 详细结果列表 */
  results: Array<{
    /** 资源ID */
    resourceId: number
    /** 是否成功 */
    success: boolean
    /** 错误信息 */
    error?: string
  }>
}

/**
 * 章节内容查询请求
 */
export interface ChapterResourceQueryDTO {
  /** 资料ID */
  resourceId?: number
  /** 所属章节ID */
  chapterId?: number
  /** 资料名称 */
  resourceName?: string
  /** 资料类型（video视频/document文档/text文本/image图片/audio音频） */
  resourceType?: 'video' | 'document' | 'text' | 'image' | 'audio' | 'archive' | 'binary'
  /** 是否可见（Y/N） */
  isVisible?: 'Y' | 'N'
  /** 状态（0正常 1停用 2已删除） */
  status?: '0' | '1' | '2'
  /** 创建开始时间 */
  beginTime?: string
  /** 创建结束时间 */
  endTime?: string
  /** 页码 */
  page?: number
  /** 每页数量 */
  size?: number
}

/**
 * 创建章节内容请求
 */
export interface ChapterResourceCreateDTO {
  /** 所属章节ID */
  chapterId: number
  /** 资料名称 */
  resourceName: string
  /** 资料类型（video视频/document文档/text文本/image图片/audio音频/archive压缩包/binary二进制文件） */
  resourceType: 'video' | 'document' | 'text' | 'image' | 'audio' | 'archive' | 'binary'
  /** 文件ID（引用sys_upload.file_id） */
  fileId?: number
  /** 外部链接URL */
  resourceUrl?: string
  /** 描述 */
  description?: string
  /** 扩展数据（JSONB格式，存储视频时长、文档页数等元数据） */
  resourceData?: Record<string, any>
  /** 显示顺序 */
  displayOrder?: number
  /** 是否可见（Y/N） */
  isVisible?: 'Y' | 'N'
}

/**
 * 更新章节内容请求
 */
export interface ChapterResourceUpdateDTO {
  /** 资料ID */
  resourceId: number
  /** 资料名称 */
  resourceName?: string
  /** 资料类型（video视频/document文档/text文本/image图片/audio音频） */
  resourceType?: 'video' | 'document' | 'text' | 'image' | 'audio' | 'archive' | 'binary'
  /** 文件ID（引用sys_upload.file_id） */
  fileId?: number
  /** 外部链接URL */
  resourceUrl?: string
  /** 描述 */
  description?: string
  /** 扩展数据（JSONB格式） */
  resourceData?: Record<string, any>
  /** 显示顺序 */
  displayOrder?: number
  /** 是否可见（Y/N） */
  isVisible?: 'Y' | 'N'
  /** 状态（0正常 1停用 2已删除） */
  status?: '0' | '1' | '2'
}

/**
 * 章节内容详细信息
 */
export interface ChapterResourceDetailVO {
  /** 资料ID */
  resourceId: number
  /** 所属章节ID */
  chapterId: number
  /** 资料名称 */
  resourceName: string
  /** 资料类型（video视频/document文档/text文本） */
  resourceType: string
  /** 文件ID（引用sys_upload.file_id） */
  fileId?: number
  /** 外部链接URL */
  resourceUrl?: string
  /** 描述 */
  description?: string
  /** 扩展数据（JSONB格式，存储视频时长、文档页数等元数据） */
  resourceData?: Record<string, any>
  /** 纯文本文件ID（PDF 解析后） */
  textFileId?: number
  /** 解析状态（0待处理 1处理中 2处理成功 3处理失败） */
  parseStatus?: string
  /** 显示顺序 */
  displayOrder: number
  /** 是否可见（Y/N） */
  isVisible: string
  /** 状态（0正常 1停用 2已删除） */
  status: string
  /** 创建者 */
  createBy?: number
  /** 创建时间 */
  createTime?: string
  /** 更新者 */
  updateBy?: number
  /** 更新时间 */
  updateTime?: string
  /** 文件URL */
  fileUrl?: string
  /** 文件上传信息（sys_upload） */
  fileInfo?: import('../system/upload').FileInfoVO
}

/**
 * 章节内容列表项
 */
export interface ChapterResourceListVO {
  /** 资料ID */
  resourceId: number
  /** 所属章节ID */
  chapterId: number
  /** 资料名称 */
  resourceName: string
  /** 资料类型（video视频/document文档/text文本） */
  resourceType: string
  /** 文件ID（引用sys_upload.file_id） */
  fileId?: number
  /** 外部链接URL */
  resourceUrl?: string
  /** 描述 */
  description?: string
  /** 扩展数据（JSONB格式） */
  resourceData?: Record<string, any>
  /** 解析状态（0待处理 1处理中 2处理成功 3处理失败） */
  parseStatus?: string
  /** 显示顺序 */
  displayOrder: number
  /** 是否可见（Y/N） */
  isVisible: string
  /** 状态（0正常 1停用 2已删除） */
  status: string
  /** 创建时间 */
  createTime?: string
  /** 文件URL */
  fileUrl?: string
  /** 文件上传信息（sys_upload） */
  fileInfo?: import('../system/upload').FileInfoVO
}

/**
 * 文档任务状态 VO（PDF 解析 / GraphRAG 构建）
 * 对应后端：章节资料解析和 GraphRAG 构建任务
 */
export interface DocumentTaskStatusVO {
  /** 文档ID */
  documentId: number
  /** PDF 解析状态：0待处理 / 1处理中 / 2处理成功 / 3处理失败 */
  parseStatus: string
  /** MinerU 任务 ID */
  mineruTaskId?: string
  /** 解析页数 */
  pageCount?: number
  /** Markdown 内容长度 */
  markdownLength?: number
  /** 纯文本文件ID */
  textFileId?: number
  /** Markdown 在对象存储中的 Key */
  markdownS3Key?: string
  /** Markdown 结果临时访问链接 */
  markdownUrl?: string
  /** 实体数量 */
  entityCount?: number
  /** 关系数量 */
  relationCount?: number
  /** 社区摘要数量 */
  communityCount?: number
  /** 错误信息（失败时） */
  errorMessage?: string
}

/**
 * PDF 解析提交响应 VO
 * 对应后端：POST /education/resource-document/{id}/parse
 */
export interface ParseSubmitVO {
  /** 文档 ID */
  documentId: number
  /** MinerU 任务 ID */
  mineruTaskId: string
  /** 解析状态，固定为 '1'(处理中) */
  parseStatus: string
}

/**
 * Celery 任务提交响应 VO
 * 用于 build-graph 等异步任务的提交结果
 */
export interface CeleryTaskVO {
  /** Celery 任务 ID */
  celeryTaskId: string
  /** 提交状态，固定为 'submitted' */
  status: string
  /** 关联文档 ID（build-graph 时） */
  documentId?: number
  /** 关联章节 ID（generate-description 时） */
  chapterId?: number
}
